conf/cmake/common.cmake

```diff
@@ -9,76 +9,87 @@ set(AREG_AR          "${CrossCompile}ar")
 set(AREG_OS          "${OpSystem}")
 set(AREG_STATIC_LIB)
 
-if(areg MATCHES "static")
-    set(AREG_BINARY "static")
-else()
+if(areg MATCHES "shared")
     set(AREG_BINARY "shared")
+else()
+    set(AREG_BINARY "static")
 endif()
 
-
 # Add compiler flags here
-add_compile_options(-g -pthread -Werror)
-
 # Checking Compiler for adding corresponded tweaks and flags
 if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
     # Clang compile options
-    add_compile_options(-stdlib=libc++)
+    list(APPEND CompileOptions -g -pthread -Werror -stdlib=libc++ ${UserDefines})
+    if (Config MATCHES "Release")
+        list(APPEND CompileOptions -O2)
+    endif()
+
 elseif (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
     # GNU compile options
-    add_compile_options(-Wall ${UserDefines})
+    list(APPEND CompileOptions -g -pthread -Werror -Wall ${UserDefines})
+    if (Config MATCHES "Release")
+        list(APPEND CompileOptions -O2)
+    endif()
+
 elseif (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
     # Visual Studio C++
+    if(Config MATCHES "Release")
+        add_definitions(-DNDEBUG -DUNICODE)
```

Comment:
I think DNDEBUG/DDEBUG is being taken care of with a specific statement.
If so, DUNICODE can be unconditionally defined here, and only define extra optimization flags in case of Release